File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = 0
        Sum = 0
        length = 9999999
        for right in range(0,len(nums)):
            Sum += nums[right]
            while Sum >= target:
                currLength = right - left + 1
                length = min(length, currLength)
                Sum -= nums[left]
                left+=1
        if length == 9999999:
            return 0
        else:
            return length

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.cs
public class Solution {
    public int MinSubArrayLen(int target, int[] nums) {
        int left  = 0;
        int currSum = 0;
        int length = int.MaxValue;

        for(int right = 0 ; right < nums.Length;right++){
            currSum += nums[right];
            while(currSum >= target){
                int currLength = right - left + 1;
                length = Math.Min(length, currLength);
                currSum -= nums[left];
                left++;
            }
        }
        return length != int.MaxValue? length:0;
    }
}